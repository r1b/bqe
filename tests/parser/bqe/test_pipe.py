import pytest

@pytest.mark.parametrize("sql", (
    pytest.param("FROM table1", id="pipe_from"),
    pytest.param("FROM table1 |> SELECT col1", id="pipe_from_select"),
    pytest.param("SELECT col1 FROM table1 |> SELECT col1", id="pipe_select_select"),
    pytest.param("(SELECT col1 FROM table1) |> SELECT col1", id="pipe_subquery_select"),
))
def test_pipe_ok(sql, assert_parse_tree):
    assert_parse_tree(sql)
