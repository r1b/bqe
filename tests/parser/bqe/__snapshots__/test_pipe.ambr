# serializer version: 1
# name: test_pipe_ok[pipe_from]
  '''
  start
    query_statement
      query_expr
        pipe_from
          from_clause
            bigquery__from_item_table
              bigquery__path_expression
                bigquery__ident	table1
  
  '''
# ---
# name: test_pipe_ok[pipe_from_select]
  '''
  start
    query_statement
      query_expr
        pipe_expr
          pipe_start
            pipe_from
              from_clause
                bigquery__from_item_table
                  bigquery__path_expression
                    bigquery__ident	table1
          pipe_item
            pipe_operator
              pipe_select
                select_list
                  bigquery__select_item
                    bigquery__select_expr
                      bigquery__column_expr
                        bigquery__ident	col1
  
  '''
# ---
# name: test_pipe_ok[pipe_select_select]
  '''
  start
    query_statement
      query_expr
        pipe_expr
          pipe_start
            select
              select_list
                bigquery__select_item
                  bigquery__select_expr
                    bigquery__column_expr
                      bigquery__ident	col1
              bigquery__select_from
                from_clause
                  bigquery__from_item_table
                    bigquery__path_expression
                      bigquery__ident	table1
          pipe_item
            pipe_operator
              pipe_select
                select_list
                  bigquery__select_item
                    bigquery__select_expr
                      bigquery__column_expr
                        bigquery__ident	col1
  
  '''
# ---
# name: test_pipe_ok[pipe_subquery_select]
  '''
  start
    query_statement
      query_expr
        pipe_expr
          pipe_start
            subquery
              query_expr
                select
                  select_list
                    bigquery__select_item
                      bigquery__select_expr
                        bigquery__column_expr
                          bigquery__ident	col1
                  bigquery__select_from
                    from_clause
                      bigquery__from_item_table
                        bigquery__path_expression
                          bigquery__ident	table1
          pipe_item
            pipe_operator
              pipe_select
                select_list
                  bigquery__select_item
                    bigquery__select_expr
                      bigquery__column_expr
                        bigquery__ident	col1
  
  '''
# ---
